### DISCUSSION TITLE
Update Interval (Getting Http 429 Response)

### DISCUSSION THREAD

**Discussion by WillScott73** [tillsteinbach/VWsFriend#565] [Q&A]
What should the ideal update interval be?

Things were working fine on 180 seconds until VW pulled We Connect in favour of their new app.

Now I'm getting back a message saying that my client has made too many requests, with an http status code of 429.

This has caused the VW app to stop working too.

**Comment by tillsteinbach** [comment]
Yes, this is a struggle! I have not found out yet. I believe this will also happen if you open the app on two phones at the same time. But I have not further investigated. If anyone has good hints I would be very interested!

**Reply by WillScott73** [reply]
Do I need to contact Volkswagen to get access to the data again, or does access reset after a given time period?

I stopped vwsfriend from polling 6 hours ago but just got the same 429 response when trying again.

**Reply by tillsteinbach** [reply]
For me it fixed itself.

**Reply by WillScott73** [reply]
It's fixed itself again for me now, so the rate limit period is at least 12 hours.

From the perspective of vwsfriend, is it preferable to update every 180 seconds? Do we miss out on data if it's less frequent.

I'll start with 600 and gradually reduce it. 

My interest is more in the overall journey data, and whether the car is locked, than anything else.

**Reply by tillsteinbach** [reply]
The Trips might be less accurate and you may loose very short trips. Apart from that no problem.

**Comment by WillScott73** [comment]
Thanks for putting in the effort to author vwsfriend, by the way.

I've been writing code since I was aged 10, I'm 50 now. I can't be bothered to write code as a hobby because I do it all day every day as my job, and have done so for 30 years now, mostly Visual Basic, C# and Golang, but I can understand Python and other languages too.

I think that it's because I'm autistic that I love collecting data, such as that provided by vwsfriend.

So thank you, once again.
